### DISCUSSION TITLE
Self-hosting and engine

### DISCUSSION THREAD

**Discussion by SlashScreen** [CanadaHonk/porffor#174] [General]
I am wondering if porffor could self-host in the future? As in, compile *itself* to wasm. If self-hosted and paired with a WASM engine, would it theoretically be capable of facilitating a "javascript engine"? As in, get javascript from a web server, compile it to WASM, and then hand it to the WASM engine to execute?

This is my odd use case: last year, I got Wazero running on the Plan 9 operating system. It's not always easy to get something like V8 or JavaScriptCore running on 9 since it isn't written in C95. Go, however, *can* build for the platform, hence why wazero is able to run. My initial thought was, if Porffor is one day able to self-host, it could be paired with wazero to create a JS runtime for Plan 9 (however slow it might end up being).

If this isn't possible, that's fine, I'm mostly just brainstorming here.

**Comment by CanadaHonk** [comment]
This is definitely a want but also quiet far away.

**Reply by SlashScreen** [reply]
is the goal to make *any* website's javascript work? So long as it's not... generated, I guess.

**Reply by CanadaHonk** [reply]
Any JS generally, hopefully, not specifically website's.

**Comment by qxrein** [comment]
Your idea is really creative! Self-hosting Porffor and compiling it to WASM could potentially make the setup you’re describing for a JavaScript runtime on Plan 9 possible. I totally agree, it’s something to look forward to, but it does seem like it’s quite a way off. Still, it's exciting to think about the potential! I would love to catch opportunity to contribute to it
